OneSchedule.Domain.Abstractions/IService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace OneSchedule.Domain.Abstractions
+{
+    public interface IService<T>
```

Comment:
You have async methods in Repository. Why does this service use synchronous approach?